### PR TITLE
XPU QLoRA Container

### PR DESCRIPTION
**Pull request by Uxito-Ada**
## Description

docker support for XPU user to run QLoRA (quantized LoRA fine-tuning)

### 1. Why the change?

a docker solution is required for XPU fine-tuning

### 2. User API changes

none

### 3. Summary of the change 

as above

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [x] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

File: docker/llm/finetune/xpu/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    apt-get install -y curl wget git gnupg gpg-agent && \
+    wget -qO - https://repositories.intel.com/graphics/intel-graphics.key | gpg --dearmor --output /usr/share/keyrings/intel-graphics.gpg && \
+    echo 'deb [arch=amd64,i386 signed-by=/usr/share/keyrings/intel-graphics.gpg] https://repositories.intel.com/graphics/ubuntu jammy arc' | tee /etc/apt/sources.list.d/intel.gpu.jammy.list && \
+    rm /etc/apt/sources.list.d/intel-graphics.list && \
+    # install python 3.9
+    ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone && \
+    env DEBIAN_FRONTEND=noninteractive apt-get update && \
+    apt install software-properties-common libunwind8-dev vim less -y && \
+    add-apt-repository ppa:deadsnakes/ppa -y && \
+    apt-get install -y python3.9 git curl wget && \
```
Comment (glorysdj) [inline docker/llm/finetune/xpu/docker/Dockerfile:18]: wget, curl, git are duplicated with line 9
please use one command to update apt-get, 3 separated commands to intsall basic lib, gpu driver and py 39

File: docker/llm/finetune/xpu/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    rm /usr/bin/python3 && \
+    ln -s /usr/bin/python3.9 /usr/bin/python3 && \
+    ln -s /usr/bin/python3 /usr/bin/python && \
+    apt-get install -y python3-pip python3.9-dev python3-wheel python3.9-distutils && \
+    curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && \
+    # install XPU bigdl-llm
+    pip install --pre --upgrade bigdl-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu && \
+    # install huggingface dependencies
+    pip install git+https://github.com/huggingface/transformers.git@95fe0f5 && \
```
Comment (glorysdj) [inline docker/llm/finetune/xpu/docker/Dockerfile:27]: make this git commit an ENV for future change?

File: docker/llm/finetune/xpu/docker/Dockerfile (hunk 1 above)
Comment (Uxito-Ada) [reply docker/llm/finetune/xpu/docker/Dockerfile]: done

File: docker/llm/finetune/xpu/docker/Dockerfile (hunk 2 above)
Comment (Uxito-Ada) [reply docker/llm/finetune/xpu/docker/Dockerfile]: done

File: docker/llm/finetune/xpu/docker/qlora_finetune_xpu.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import torch
```
Comment (yangw1234) [inline docker/llm/finetune/xpu/docker/qlora_finetune_xpu.py:1]: Is it possible to use the original example. 

I am afraid this one will be easily forgot when we do future update.

**Review by yangw1234 — APPROVED** [review]
LGTM

File: docker/llm/finetune/xpu/docker/qlora_finetune_xpu.py (hunk 3 above)
Comment (Uxito-Ada) [reply docker/llm/finetune/xpu/docker/qlora_finetune_xpu.py]: done

**Pull request merged**